node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Nodes.java

```diff
@@ -641,7 +641,9 @@ private void requireBreakfixable(Node node) {
             Node newHost = host.withWantToRetire(true, wantToDeprovision, wantToRebuild, agent, instant);
             if (op == DecommissionOperation.encrypt) {
                 Report report = Report.basicReport(Report.WANT_TO_ENCRYPT_ID, Report.Type.UNSPECIFIED, instant, "");
-                newHost = newHost.with(newHost.reports().withReport(report));
+                newHost = newHost.with(newHost.reports().withReport(report))
+                                 // Restart host-admin to trigger run of single-resume disk tasks
+                                 .withRestart(newHost.allocation().get().restartGeneration().withIncreasedWanted());
```

Comment:
The restart must be triggered only after it reaches "parked".

Reply:
Of course, thanks!